src/plone/restapi/serializer/converters.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+from datetime import date
+from datetime import datetime
+from DateTime import DateTime
+from datetime import time
+from persistent.list import PersistentList
+from persistent.mapping import PersistentMapping
+from plone.app.textfield.interfaces import IRichTextValue
+from plone.restapi.interfaces import IJsonCompatible
+from Products.CMFPlone.utils import getSiteEncoding
+from Products.CMFPlone.utils import safe_unicode
+from zope.component import adapter
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+
+
+def json_compatible(value):
+    """The json_compatible function converts any value to JSON compatible
+    data when possible, raising a TypeError for unsupported values.
+    This happens by using the IJsonCompatible.
+
+    Be aware that adapting the value `None` will result in a component
+    lookup error unless `None` is passed in as default value.
+    Because of that the `json_compatible` helper method should always be
+    used for converting values that may be None.
+    """
+    return IJsonCompatible(value, None)
+
+
+@adapter(Interface)
+@implementer(IJsonCompatible)
+def default_converter(value):
+    if value is None:
+        return value
+
+    if type(value) in (unicode, bool, int, float, long):
```

Comment:
Hmm. What about `isinstance(value, (unicode, bool, int, float, long))`? The `json` module seems to be able to serialize subclasses of those types just fine:

```python
>>> import json
>>>
>>> class MyUnicode(unicode): pass
...
>>> value = MyUnicode('foobar')
>>> json.dumps(value)
'"foobar"'
```